tests/functional/commands/test_insert_command.py

```diff
@@ -58,24 +56,24 @@ def test_insert_single_trial(database, monkeypatch, script_path):
         ]
     )
 
-    exp = list(database.experiments.find({"name": "test_insert_normal"}))
+    exp = list(storage.fetch_experiments({"name": "test_insert_normal"}))
 
     assert len(exp) == 1
     exp = exp[0]
     assert "_id" in exp
 
-    trials = list(database.trials.find({"experiment": exp["_id"]}))
+    trials = list(storage._fetch_trials({"experiment": exp["_id"]}))
```

Comment:
```suggestion
    trials = list(storage.fetch_trials(uid=exp["_id"]))
```

Reply:
On pourrait aussi faire `storage.fetch_trials(experiment=exp)`, right?

Reply:
Pas si c'est un dict.

Reply:
Ah, true. But why does `fetch_experiments` return dicts rather than `Experiment` objects? `fetch_trials` does return `Trial` objects.

Reply:
Yeah it is inconsistent. It is because we need to make more complex operations after fetching the experiment config in order to instantiate it.

Reply:
Okay, I see. What kind of operations?

Reply:
We fetch the experiment config, then compare the config from DB with the config passed by user and detect if there is any modification. If there is we branch and create a new experiment that we register in the database.

Reply:
We are planning to divide the experiment class into 2, one being a data container and the other being the client object giving access to operations with the database (like selecting trials, creating trials). When this is done we could make fetch_experiment returns the experiment data containers, just like it is done for the trials.